server/server/src/billing/billing_service.rs

```diff
@@ -45,37 +46,42 @@ fn lock_subscription_profile(
     state: &ServerState, public_key: &PublicKey,
 ) -> Result<Account, ServerError<LockBillingWorkflowError>> {
     let owner = Owner(*public_key);
-    state.index_db.transaction(|tx| {
-        let mut account = tx
-            .accounts
-            .get(&owner)
-            .ok_or(ClientError(UserNotFound))?
-            .clone();
-        let current_time = get_time().0 as u64;
-
-        if current_time - account.billing_info.last_in_payment_flow < state.config.billing.millis_between_user_payment_flows {
-            warn!(?owner, "User/Webhook is already in payment flow, or not enough time that has elapsed since a failed attempt");
-
-            return Err(ClientError(ExistingRequestPending));
-        }
+    let mut db = state.index_db.lock()?;
+    let tx = db.begin_transaction()?;
+    let mut account = db
+        .accounts
+        .remove(&owner)?
```

Comment:
Should this `remove` be a `get`? There are a few ways for this function to early terminate; is the transaction automatically aborted or something like that?

Reply:
[215f6d8](https://github.com/lockbook/lockbook/pull/1588/commits/215f6d8ce2369a02b2aaf3a69aa778be19c770ef)